hal/network/ncp_client/quectel/quectel_ncp_client.cpp

```diff
@@ -368,7 +368,7 @@ int QuectelNcpClient::off() {
         // WARN: We assume that the modem can turn off itself reliably.
     } else {
         // Power down using hardware
-        modemPowerOff();
+        CHECK(modemPowerOff());
```

Comment:
This leaves the NCP client in a half-initialized state. Any external interaction with it may bring it into an on state again (see `waitReady()`). If the idea is to make sure we are powered off, we should rely on the power state to decide that we should call `off()` again.